src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/canceltranslations/CancelTranslationsAboutToSubmitHandlerTest.java

```diff
@@ -59,120 +61,128 @@ public void givenCanHandleIsCalledWithValidCallBack_shouldReturnTrue() {
-        assertNull(response.getData().getSscsDocument().get(4).getValue().getDocumentTranslationStatus());
-        assertNull(response.getData().getSscsDocument().get(5).getValue().getDocumentTranslationStatus());
-        assertNull(response.getData().getSscsDocument().get(6).getValue().getDocumentTranslationStatus());
+            handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+        if (sscsDocumentNumber == 3) {
+            assertEquals(SscsDocumentTranslationStatus.TRANSLATION_COMPLETE,
+                response.getData().getSscsDocument().get(0).getValue().getDocumentTranslationStatus());
+        } else {
+            assertNull(response.getData().getSscsDocument().get(0).getValue().getDocumentTranslationStatus());
+        }
         assertEquals("No", response.getData().getTranslationWorkOutstanding());
 
         assertEquals(REVIEW_BY_TCW, response.getData().getInterlocReviewState());
     }
 
-    private Callback<SscsCaseData> buildCallback(EventType eventType, State state) {
-
-        SscsDocument ssc0Doc =
-                buildSscsDocument("english.pdf", SscsDocumentTranslationStatus.TRANSLATION_REQUIRED,
-                        DocumentType.APPELLANT_EVIDENCE.getValue(), LocalDate.now());
-
-        SscsDocument sscs1Doc =
-                buildSscsDocument("english.pdf", SscsDocumentTranslationStatus.TRANSLATION_REQUIRED,
-                        DocumentType.SSCS1.getValue(), LocalDate.now().minusDays(1));
-
-        SscsDocument sscs2Doc =
-                buildSscsDocument("anything.pdf", SscsDocumentTranslationStatus.TRANSLATION_REQUIRED,
-                        DocumentType.DECISION_NOTICE
-                                .getValue(), LocalDate.now().minusDays(3));
-
-        SscsDocument sscs3Doc =
-                buildSscsDocument("anything.pdf", SscsDocumentTranslationStatus.TRANSLATION_COMPLETE,
-                        DocumentType.DIRECTION_NOTICE
-                                .getValue(), LocalDate.now().minusDays(2));
-
-        SscsDocument sscs4Doc =
-                buildSscsDocument("anything.pdf", null, DocumentType.DECISION_NOTICE
-                        .getValue(), LocalDate.now().minusDays(2));
-
-        SscsDocument sscs5Doc =
-                buildSscsDocument("anything.pdf", SscsDocumentTranslationStatus.TRANSLATION_REQUESTED,
-                        DocumentType.DIRECTION_NOTICE
-                                .getValue(), LocalDate.now().minusDays(2));
-
-        SscsDocument sscs6Doc =
-                buildSscsDocument("anything.pdf", SscsDocumentTranslationStatus.TRANSLATION_REQUIRED,
-                        DocumentType.REINSTATEMENT_REQUEST
-                                .getValue(), LocalDate.now().minusDays(2));
-
-        List<SscsDocument> docs = Arrays.asList(ssc0Doc, sscs1Doc, sscs2Doc, sscs3Doc, sscs4Doc, sscs5Doc, sscs6Doc);
+    private Callback<SscsCaseData> buildCallback(EventType eventType, State state, int sscsDocumentNumber) {
```

Comment:
I'm not 100% sure what's going on here but could you use a switch case (pattern matching) on the EventType or State instead of an index?

Reply:
Nope not really, the index is for like document types and scenarios, but the state and event type are basically the same per overarching test

Reply:
good shout, have done so